### PR TITLE
Support for http and other filesystems in path of data source

### LINKED ISSUE #499: Adding support loading data from http[s]
### Kind of request

Adding new functionality

### Enhancement Description

Hello! Loading data from https is not supported, but is for example in rioxarray. This would be great to support, similar to s3. For example, the following datacatalog.yml specifies the path as https. This https url leads directly to a 13GB tif file of wind speed. 

```
global_wind_atlas:
  crs: 4326
  data_type: RasterDataset
  driver: raster
  path: https://globalwindatlas.info/api/gis/global/wind-speed/10
  meta:
    category: climate
    source_license: CC BY 4.0
    source_url: https://globalwindatlas.info
```

Loading this data in rioxarray works using the path of the entry in the data catalog. It is loaded within seconds and I can use (for example) `rio.clip_bbox` to select and download only the data that I want.

```
rxr.open_rasterio(self.data_catalog['global_wind_atlas'].path)
> xarray.DataArray ...
```

However, when doing the same thing directly using the ```get_raterdataset``` method of the data catalog, the following error occurs in `resolve_paths`: 
```
self.data_catalog.get_rasterdataset('global_wind_atlas')
> No such file found: https://globalwindatlas.info/api/gis/global/wind-speed/10
```

### Use case

Loading data remotely from https

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by DirkEilander**
## Issue addressed
Fixes #499 

## Explanation
The `filesystem` property is now detected from the path (and hence optional in the data catalog).
The `storage_options` argument is set outside the `driver_kwargs` argument as it is not passed to the driver, but to the fsspec filesystems method. (needs to be updated in future data catalogs, now backwards compatible).

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.rst if needed

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

**Comment by DirkEilander** [comment]
@savente93 Any clue why this test would fail? It runs locally at my machine..

**Comment by DirkEilander** [comment]
One ToDo left regarding docs, but the code can reviewed if anyone has time.

**Comment by savente93** [comment]
The process exited midway because of a "graceful shutdown signal" from the os. I.e. "You're misbehaving so we're terminating you, but you get the opportunity to clean up after yourself". So this is likely either a resource or permission error. I've restarted the tests to see if the issue persists. If it's still there after lunch I'll try and investigate

**Comment by savente93** [comment]
update: Process seems to be killed due to out of memory errors, something is hogging all the memory. My guess is that somewhere there's an accidental infinite loop or infinite recursion where thengs get allocated

**Comment by savente93** [comment]
Issue seems to be resolved now for whatever reason so leaving it here. will review the code in a moment.

File: hydromt/data_adapter/data_adapter.py [hunk 1]
```diff
@@ -280,95 +304,82 @@ def _resolve_paths(
...
-            else:
-                raise ModuleNotFoundError(
-                    "The s3fs library is required to read data from s3"
-                    + " (Amazon Web Storage). Please install."
-                )
+        if len(keys) > 0:
+            # resolve dates: month & year keys
+            if time_tuple is not None:
+                dt = pd.to_timedelta(self.unit_add.get("time", 0), unit="s")
+                trange = pd.to_datetime(list(time_tuple)) - dt
```
[the remark this answers is not in the document]
Comment (DirkEilander) [reply hydromt/data_adapter/data_adapter.py]: done

**Review by savente93 — APPROVED** [review]
all good now!

**Pull request merged**